src/core/jit/synth/synth.rs

```diff
@@ -0,0 +1,426 @@
+use async_graphql::Positioned;
+
+use crate::core::jit::model::{Field, Nested, OperationPlan, Variable, Variables};
+use crate::core::jit::store::{Data, DataPath, Store};
+use crate::core::jit::synth::Synthesizer;
+use crate::core::jit::{Error, ValidationError};
+use crate::core::json::{JsonLikeOwned, JsonObjectLike};
+use crate::core::scalar::get_scalar;
+
+// TODO: rename
+pub struct AlsoSynth<Value> {
+    plan: OperationPlan<Value>,
+}
+
+impl<Value: JsonLikeOwned> AlsoSynth<Value> {
+    pub fn new(plan: OperationPlan<Value>) -> Self {
+        Self { plan }
+    }
+}
+
+impl<Value: JsonLikeOwned + Clone> Synthesizer for AlsoSynth<Value> {
```

Comment:
Now that Synth has been parameterized, we don't need the `Synthesizer` trait anymore.